### PR TITLE
chore(dashboard): update TiDB Dashboard to v6.5.11-9d2b0ce6 [release-6.5]

### PR DESCRIPTION
**Pull request by baurine**
### What problem does this PR solve?

Issue Number: ref #8611.

Update TiDB Dashboard to [v6.5.11-9d2b0ce6](https://github.com/pingcap/tidb-dashboard/releases/tag/v6.5.11-9d2b0ce6).

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by mornyx** [comment]
This is a 6.5 cherry-pick of https://github.com/tikv/pd/pull/8581, PTAL, thx!
/cc @lhy1024 @rleungx @niubell

File: pkg/dashboard/adapter/config.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ func GenDashboardConfig(srv *server.Server) (*config.Config, error) {
 	dashboardCfg.PublicPathPrefix = cfg.Dashboard.PublicPathPrefix
 	dashboardCfg.EnableTelemetry = cfg.Dashboard.EnableTelemetry
 	dashboardCfg.EnableExperimental = cfg.Dashboard.EnableExperimental
+	dashboardCfg.DisableCustomPromAddr = cfg.Dashboard.DisableCustomPromAddr
```
Comment (rleungx) [inline pkg/dashboard/adapter/config.go:38]: Will the config be lost when we upgrade?

Comment (mornyx) [reply pkg/dashboard/adapter/config.go:38]: > Will the config be lost when we upgrade?

This configuration already exists in the `master` and `release-6.5` branches of tidb-dashboard and will not be lost during a normal upgrade.

Comment (rleungx) [reply pkg/dashboard/adapter/config.go:38]: What if we set it to true in 6.5.11, then the cluster is upgraded to 7.1.x which doesn't contain this PR?

Comment (mornyx) [reply pkg/dashboard/adapter/config.go:38]: Oh, I get what you mean. This change will be cherry-picked to all LTS branches later. (At least for now only 6.5.x customers explicitly need to configure this.)

Comment (mornyx) [reply pkg/dashboard/adapter/config.go:38]: If an upgrade occurs during this period, this configuration will indeed lose its effect.

Comment (rleungx) [reply pkg/dashboard/adapter/config.go:38]: After losing it, if we upgrade again to a higher version with this config, the config value will be changed to false.

Comment (mornyx) [reply pkg/dashboard/adapter/config.go:38]: I think this is acceptable and I will doc this behavior. It is expected that this configuration will not work after upgrading to a version that does not support this feature.

**Review by rleungx — APPROVED** [review]

**Review by HuSharp — APPROVED** [review]

**Pull request merged**